### ISSUE TITLE
Dynamic discrete actions

### ISSUE THREAD

**Issue by windmaple** [reinforceio/tensorforce#62]
Hi,

How could one dynamically reduce action space? For example, in Go, the agent's action are gradually shrunk since it shouldn't place stones in a repeat position. It doesn't seem we can do this w/ TensorFoce at the moment?

Obviously I thought about making any repeat action prohibitively expensive in the env, but I'm just curious if there is another way.

**Comment by Islandman93** [comment]
@windmaple The network will always have a fixed number of outputs so you wouldn't be able to do this in plain TensorFlow either. Normally you would just return the probabilities from a softmax layer, and if the maximum is not a valid move try the next highest value and so on.

**Comment by michaelschaarschmidt** [comment]
Correct, and we do not have any plans to support dynamic computation graphs unless for custom projects. Constraining the action space is done on the application side
